GitUI/GitUIExtensions.cs

```diff
@@ -115,10 +115,15 @@ void OpenWithDiffTool()
                 if (file.IsSubmodule)
                 {
                     // Patch already evaluated
```

Comment:
Move comment to inside `if (task`

or rather move all submodule handling here

something like
````c#
if (task is null)
{
    var patch = GetItemPatch(fileViewer.Module, file, firstId, selectedId,
                    fileViewer.GetExtraDiffArguments(), fileViewer.Encoding);
    return LocalizationHelpers.ProcessSubmodulePatch(fileViewer.Module, file.Name, patch);
}
else
{
    // Patch already evaluated
    var status = ThreadHelper.JoinableTaskFactory.Run(() => task);
    return status is not null
                            ? LocalizationHelpers.ProcessSubmoduleStatus(fileViewer.Module, status)
                            : $"Failed to get status for submodule \"{file.Name}\"";
                    }
```

Reply:
Some good use of pattern matching declarations...

```c#
                if (file.IsSubmodule)
                {
                    if (file.GetSubmoduleStatusAsync() is not Task<GitSubmoduleStatus> task)
                    {
                        var diff = GetItemPatch(fileViewer.Module, file, firstId, selectedId,
                            fileViewer.GetExtraDiffArguments(), fileViewer.Encoding);
                        return LocalizationHelpers.ProcessSubmodulePatch(fileViewer.Module, file.Name, diff);
                    }

                    // Patch already evaluated
                    var status = ThreadHelper.JoinableTaskFactory.Run(() => task);
                    return status is not null
                        ? LocalizationHelpers.ProcessSubmoduleStatus(fileViewer.Module, status)
                        : $"Failed to get status for submodule \"{file.Name}\"";
                }
```